code/datums/components/anti_magic.dm

```diff
@@ -4,41 +4,14 @@
 	var/holy = FALSE
 
 /datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(COMSIG_ITEM_USER_RECEIVE_MAGIC, .proc/can_protect)
+
 	magic = _magic
 	holy = _holy
 
 /datum/component/anti_magic/proc/can_protect(_magic = TRUE, _holy = FALSE)
-	if(!active)
```

Comment:
either this check should not be dropped or the `active` var should be removed

Reply:
I thought this was the component var for if the component should be enabled but it turns out it's just more duplicated functionality.

Reply:
This var is so you can turn off the antimagic without having to add/remote the component itself

Reply:
Components already have a var that does that

Reply:
Oh, right yeah now that it works with signals you can use the default var